cmd/utils/cmd.go

```diff
@@ -262,6 +264,96 @@ func ExportChain(blockchain *core.BlockChain, fn string) error {
 	return nil
 }
 
+func numberedFileName(basename string, n int) string {
+	suffix := filepath.Ext(basename)
+	name := strings.TrimSuffix(basename, suffix)
+	name = name + fmt.Sprintf("-%d", n) + suffix
+	return name
+}
+
+func ExportHistory(bc *core.BlockChain, fn string, targetSize int) error {
+	return ExportHistoryRange(bc, fn, uint64(0), bc.CurrentBlock().NumberU64(), targetSize)
+}
+
+func ExportHistoryRange(bc *core.BlockChain, fn string, first uint64, last uint64, targetSize int) error {
```

Comment:
I think there should be verification that `first+last < maxBlocks`?

Reply:
🤔 what would maxBlocks here here?